components/list/list-item-checkbox-mixin.js

```diff
@@ -59,6 +65,32 @@ export const ListItemCheckboxMixin = superclass => class extends SkeletonMixin(L
 		if (!this.key) this.setSelected(undefined, true);
 	}
 
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (this.selectable) {
+			this.shadowRoot.querySelector('slot[name="nested"]').addEventListener('slotchange', e => {
```

Comment:
Could also ask consumers to do this wire-up in their `render()`, which would ensure that the event listener gets removed.